proof/refine/RISCV64/TcbAcc_R.thy

```diff
@@ -3035,57 +3039,126 @@ lemma ko_at'_valid_tcbs'_valid_tcb':
 defs valid_tcbs'_asrt_def:
   "valid_tcbs'_asrt \<equiv> valid_tcbs'"
 
-lemma isSchedulable_corres:
+declare valid_tcbs'_asrt_def[simp]
+
+(* FIXME RT: move to Lib *)
+lemma corres_gets_return_trivial:
+  "(\<And>s s'. \<lbrakk>(s, s') \<in> sr; P s; P' s'\<rbrakk> \<Longrightarrow> f s = v)
+   \<Longrightarrow> corres_underlying sr nf nf' (=) P P' (gets f) (return v)"
+  unfolding corres_underlying_def gets_def get_def return_def bind_def
+  by clarsimp
+
+definition schedulable' :: "machine_word \<Rightarrow> kernel_state \<Rightarrow> bool" where
+  "schedulable' tcbPtr s' \<equiv>
+   (runnable' |< (tcbs_of' s' ||> tcbState)) tcbPtr
+   \<and> ((\<lambda>sc. 0 < scRefillMax sc) |< (tcbs_of' s' |> tcbSchedContext |> scs_of' s')) tcbPtr
+   \<and> (Not \<circ> tcbInReleaseQueue |< tcbs_of' s') tcbPtr"
```

Comment:
I think I have seen these terms occur somewhere in the proofs (CRefine?) -- I would still vote for giving them a name, at least the ones that are referenced again. It's not only that it makes it easier to understand, you can also click on the name and directly get to the entity that the proof makes a connection to.

Reply:
I guess that one is short enough to just leave. It's short enough , and a name for this one would require some acrobatics around `tcbInReleaseQueue`, because that is close to the name I would probably expect for it.